File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/client/rxjava3/MyService.kt
package io.micronaut.docs.client.rxjava3

//tag::class[]
import io.kubernetes.client.openapi.models.V1PodList
import io.micronaut.kubernetes.client.rxjava3.CoreV1ApiRxClient
import io.reactivex.rxjava3.core.Single
import jakarta.inject.Singleton

@Singleton
class MyService(private val coreV1ApiRxClient: CoreV1ApiRxClient) {

    fun myMethod(namespace: String) {
        val v1PodList: Single<V1PodList> = coreV1ApiRxClient.listNamespacedPod(namespace).execute()
    }
}
//end::class[]

File: test-suite-python/src/test/python/micronaut/docs/client/rxjava3/MyService.py
# tag::class[]
from jakarta.inject import Singleton
from micronaut.kubernetes.client.rxjava3 import CoreV1ApiRxClient


@Singleton
class MyService:

    def __init__(self, core_v1_api_rx_client: CoreV1ApiRxClient):
        self.core_v1_api_rx_client = core_v1_api_rx_client

    def my_method(self, namespace: str) -> None:
        v1_pod_list = self.core_v1_api_rx_client.listNamespacedPod(namespace).execute()
# end::class[]
